core/src/main/java/org/opensearch/sql/planner/physical/MLCommonsOperator.java

```diff
@@ -0,0 +1,155 @@
+package org.opensearch.sql.planner.physical;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.ml.client.MachineLearningClient;
+import org.opensearch.ml.common.dataframe.ColumnMeta;
+import org.opensearch.ml.common.dataframe.ColumnValue;
+import org.opensearch.ml.common.dataframe.DataFrame;
+import org.opensearch.ml.common.dataframe.DataFrameBuilder;
+import org.opensearch.ml.common.dataframe.Row;
+import org.opensearch.ml.common.dataset.DataFrameInputDataset;
+import org.opensearch.ml.common.parameter.FunctionName;
+import org.opensearch.ml.common.parameter.KMeansParams;
+import org.opensearch.ml.common.parameter.MLAlgoParams;
+import org.opensearch.ml.common.parameter.MLInput;
+import org.opensearch.ml.common.parameter.MLPredictionOutput;
+import org.opensearch.sql.ast.expression.Argument;
+import org.opensearch.sql.data.model.ExprDoubleValue;
+import org.opensearch.sql.data.model.ExprIntegerValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+
+/**
+ * ml-commons Physical operator to call machine learning interface to get results for
+ * algorithm execution.
+ */
+@RequiredArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class MLCommonsOperator extends PhysicalPlan {
+  @Getter
+  private final PhysicalPlan input;
+
+  @Getter
+  private final String algorithm;
+
+  @Getter
+  private final List<Argument> arguments;
+
+  @Getter
+  private final MachineLearningClient machineLearningClient;
+
+  @EqualsAndHashCode.Exclude
+  private Iterator<ExprValue> iterator;
+
+  @Override
+  public void open() {
+    super.open();
+    DataFrame inputDataFrame = generateInputDataset();
+    MLAlgoParams mlAlgoParams = convertArgumentToMLParameter(arguments.get(0), algorithm);
```

Comment:
Does it make sense to parse Argument in Analyzer?

Reply:
I prefer to put Argument parsing logic in MLCommonsOperator, because I don't see any argument parsing logic in Analyzer for other logical plans. Also, it looks like the main purpose of Analyzer class is to construct the logical plan, so I prefer to leave the argument parsing work for the actual operator, which is MLCommonsOperator.

Reply:
If we parse Argument in Analyzer, it will create dependencies in Core module as Analyzer class sits in Core module.